cedar-policy-core/src/parser/cst_to_ast.rs

```diff
@@ -4189,6 +4197,102 @@ mod tests {
         });
     }
 
+    #[test]
+    fn scope_compare_to_string() {
+        let p_src = r#"permit(principal == "alice", action, resource);"#;
+        assert_matches!(parse_policy_template(None, p_src), Err(e) => {
+            expect_err(p_src, &miette::Report::new(e), &ExpectedErrorMessageBuilder::error(
+                r#"expected an entity uid or matching template slot, found literal `"alice"`"#
+            ).help(
+                "try including the entity type if you intended this string to be an entity identifier"
```

Comment:
Nit, to match the terminology in the main error message (same comment for other messages, this was just the first place I saw it)
```suggestion
                "try including the entity type if you intended this string to be an entity uid"
```